internal/fs/fs.go

```diff
@@ -1989,27 +2028,15 @@ func (fs *fileSystem) renameDir(
 	// lookUpOrCreateChildInode (since the pending inodes are not sent back to
 	// the kernel) and unlocks the pending inodes, but only once
 	var pendingInodes []inode.DirInode
-	releaseInodes := func() {
-		for _, in := range pendingInodes {
-			fs.unlockAndDecrementLookupCount(in, 1)
-		}
-		pendingInodes = []inode.DirInode{}
-	}
-	defer releaseInodes()
+	defer fs.handlePendingInodes(&pendingInodes)()
```

Comment:
Is this working as intended. You are calling a function which is returning a function ??

Reply:
Yeah... It was working correctly...but then I realized I made it unnecessarily complex. 

Fixed.